### PR TITLE
docs(config): improve `serverMiddleware` jsdocs

### PR DESCRIPTION
**Pull request by danielroe**
### 🔗 Linked issue

resolves #4591

### ❓ Type of change

- [x] 📖 Documentation (updates to the documentation or readme)
- [ ] 🐞 Bug fix (a non-breaking change that fixes an issue)
- [ ] 👌 Enhancement (improving an existing functionality like performance)
- [ ] ✨ New feature (a non-breaking change that adds functionality)
- [ ] ⚠️ Breaking change (fix or feature that would cause existing functionality to change)

### 📚 Description

This removes some of the old data from `serverMiddleware` docs. I feel this could be considerably improved, maybe incorporating some of the nice new API docs... Or adding a notice suggesting using auto-detected `~/server/api` etc...?

cc: @clemcode 

### 📝 Checklist

- [x] I have linked an issue or discussion.
- [x] I have updated the documentation accordingly.

### CODE REVIEW

File: packages/schema/src/config/_common.ts [hunk 1]
```diff
@@ -344,27 +344,23 @@ export default {
    *   // Will register redirect-ssl npm package
    *   'redirect-ssl',
    *   // Will register file from project server-middleware directory to handle /server-middleware/* requires
-   *   { path: '/server-middleware', handler: '~/server-middleware/index.js' },
+   *   { path: '/server-middleware/**', handler: '~/server-middleware/index.ts' },
    *   // We can create custom instances too, but only in development mode, they are ignored for the production bundle.
-   *   { path: '/static2', handler: serveStatic(fileURLToPath(new URL('./static2', import.meta.url))) }
+   *   { path: '/static2/**', handler: serveStatic(fileURLToPath(new URL('./static2', import.meta.url))) }
```
Comment (pi0) [inline packages/schema/src/config/_common.ts:349]: I would rather not to advocate routing features alongside with legacy server middleware

File: packages/schema/src/config/_common.ts [hunk 2]
```diff
@@ -344,27 +344,23 @@ export default {
    *   // Will register redirect-ssl npm package
    *   'redirect-ssl',
    *   // Will register file from project server-middleware directory to handle /server-middleware/* requires
-   *   { path: '/server-middleware', handler: '~/server-middleware/index.js' },
+   *   { path: '/server-middleware/**', handler: '~/server-middleware/index.ts' },
```
Comment (pi0) [inline packages/schema/src/config/_common.ts:347]: We can omit extension for being language agnostic:

```suggestion
   *   { path: '/server-middleware/**', handler: '~/server-middleware/my-middleware' },
```

File: packages/schema/src/config/_common.ts [hunk 3]
```diff
@@ -344,27 +344,23 @@ export default {
...
    * form with a specific path.
    *
    * If you pass a string handler, Nuxt will expect that file to export a default function
-   * that handles `(req, res, next) => void`.
+   * produced with `eventHandler` from `h3`.
    *
    * @example
    * ```js
-   * export default function (req, res, next) {
+   * export default eventHandler(event => {
```
Comment (pi0) [inline packages/schema/src/config/_common.ts:361]: server-middleware option is basically a legacy option I would mark as deprecated soon. We don't need to update it's example with something it is not mean to be. (BTW for consistency we use `defineEventHandler` across Nuxt docs)

File: packages/schema/src/config/_common.ts (hunk 1 above)
Comment (danielroe) [reply packages/schema/src/config/_common.ts:349]: then let's remove this entirely, as it won't work without the change.

Comment (pi0) [reply packages/schema/src/config/_common.ts:349]: Okay right it seems a bug that we are broke with new handlers for legacy middleware.

File: packages/schema/src/config/_common.ts (hunk 3 above)
Comment (danielroe) [reply packages/schema/src/config/_common.ts:361]: Hm. I don't like leaving a legacy example. Shall we just remove and link to https://v3.nuxtjs.org/guide/features/server-routes#server-middleware instead?

Comment (pi0) [reply packages/schema/src/config/_common.ts:361]: Me either but this IS a legacy option :D Let's introduce new options and deprecate this.

File: packages/schema/src/config/_common.ts (hunk 1 above)
Comment (danielroe) [reply packages/schema/src/config/_common.ts:349]: tracked in: https://github.com/nuxt/framework/issues/4235

**Comment by pi0** [comment]
With #4850 this change is not relavant anymore as we register legacy `serverMiddleware` exactly as middleware not route handlers. Also deprecated by #4850 and removed from the docs.

**Pull request closed without merging**